src/funcCoreTools/funcHostTask.ts

```diff
@@ -5,61 +5,76 @@
+            isFuncHostRunningMap.set(e.execution.task.scope, false);
         }
     });
 
     registerEvent('azureFunctions.onDidTerminateDebugSession', vscode.debug.onDidTerminateDebugSession, stopFuncTaskIfRunning);
 }
 
-async function stopFuncTaskIfRunning(this: IActionContext): Promise<void> {
+async function stopFuncTaskIfRunning(this: IActionContext, debugSession: vscode.DebugSession): Promise<void> {
     this.suppressErrorDisplay = true;
     this.suppressTelemetry = true;
 
-    const funcExecution: vscode.TaskExecution | undefined = vscode.tasks.taskExecutions.find((te: vscode.TaskExecution) => isFuncHostTask(te.task));
-    if (funcExecution && isFuncHostRunning) {
-        this.suppressTelemetry = false; // only track telemetry if it's actually the func task
-        stopFuncHostPromise = new Promise((resolve: () => void, reject: (e: Error) => void): void => {
-            const listener: vscode.Disposable = vscode.tasks.onDidEndTask((e: vscode.TaskEndEvent) => {
-                if (isFuncHostTask(e.execution.task)) {
-                    resolve();
-                    listener.dispose();
-                }
-            });
-
-            const timeoutInSeconds: number = 30;
-            const timeoutError: Error = new Error(localize('failedToFindFuncHost', 'Failed to stop previous running Functions host within "{0}" seconds. Make sure the task has stopped before you debug again.', timeoutInSeconds));
-            setTimeout(() => { reject(timeoutError); }, timeoutInSeconds * 1000);
+    if (debugSession.workspaceFolder) {
+        const funcExecution: vscode.TaskExecution | undefined = vscode.tasks.taskExecutions.find((te: vscode.TaskExecution) => {
+            return te.task.scope === debugSession.workspaceFolder && isFuncHostTask(te.task);
         });
-        funcExecution.terminate();
-        await stopFuncHostPromise;
+
+        if (funcExecution && isFuncHostRunning(debugSession.workspaceFolder)) {
+            this.suppressTelemetry = false; // only track telemetry if it's actually the func task
+            const stopFuncHostPromise: Promise<void> = new Promise((resolve: () => void, reject: (e: Error) => void): void => {
+                const listener: vscode.Disposable = vscode.tasks.onDidEndTask((e: vscode.TaskEndEvent) => {
+                    if (e.execution === funcExecution) {
+                        resolve();
+                        listener.dispose();
+                    }
+                });
+
+                const timeoutInSeconds: number = 30;
+                const timeoutError: Error = new Error(localize('failedToFindFuncHost', 'Failed to stop previous running Functions host within "{0}" seconds. Make sure the task has stopped before you debug again.', timeoutInSeconds));
+                setTimeout(() => { reject(timeoutError); }, timeoutInSeconds * 1000);
+            });
+            stopFuncHostPromiseMap.set(debugSession.workspaceFolder, stopFuncHostPromise);
```

Comment:
What's the point of setting this here if you just await the stopFuncHostPromise right after??

Reply:
`stopFuncHostPromiseMap` is used outside this function (in `stopFuncHost` above), which could be called outside this code path